lib/cli.c

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2016-2017 Noémi Ványi
+ * Copyright (c) 2017 BalaBit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ */
+
+
+#include "cli.h"
+#include "messages.h"
+
+#include <string.h>
+
+static const char DELIMITER[] = " ";
+static const char DRIVER_TEMPLATE[] = "%s {%s};\n";
+static const char CFG_FILE_TEMPLATE[] =
+  "@version: %s\n"
+  "@include scl.conf\n"
+  "log {\n"
+  "    source {\n"
+  "        stdin();\n"
+  "    };\n"
+  "    %s"
+  "    destination {\n"
+  "        stdout(%s);\n"
+  "    };\n"
+  "};";
+
+CliParam *
+cli_param_new(gchar *type, gchar *cfg)
+{
+  CliParam *self = g_new0(CliParam, 1);
+  self->type = type;
+  self->cfg = cfg;
+  return self;
+}
+
+static gchar *
+_get_config_of_driver(CliParam *driver)
+{
+  return g_strdup_printf(DRIVER_TEMPLATE, driver->type, driver->cfg);
```

Comment:
I wouldn't use these template variable (I think the code is readable when you have the format string next to the variables).